CryptoWallet/Repositories/EthereumInteraction.swift

```diff
@@ -125,7 +125,24 @@ enum EthereumInteraction {
         let web3 = Web3.InfuraMainnetWeb3()
         do {
             let balanceResult = try web3.eth.getBalance(address: walletAddress)
-            return Web3.Utils.formatToEthereumUnits(balanceResult, toUnits: .eth, decimals: 3)
+            return Web3.Utils.formatToEthereumUnits(balanceResult, toUnits: .eth, decimals: 18)
+        } catch {
+            return nil
+        }
+    }
+
+    static func getEstimatedFee() -> (estimatedFee: Double, gasPrice: Double)? {
+        let web3 = Web3.InfuraMainnetWeb3()
+        do {
+            let gasPrice = try web3.eth.getGasPrice()
+            let fee = gasPrice * 21_000
+            guard let stringFee = Web3.Utils.formatToEthereumUnits(fee,
+                                                                   toUnits: .eth,
+                                                                   decimals: 18),
+                let doubleFee = Double(stringFee) else {
```

Comment:
```suggestion
                let fee = Double(stringFee) else {
```
để tên biến như kia đọc dễ hiểu nhầm thành nhân 2 giá lắm :v